hq/app/schedule/IamAudit.scala

```diff
@@ -6,29 +6,51 @@ import logic.DateUtils
 import model._
 import org.joda.time.{DateTime, Days}
 import play.api.Logging
-import schedule.IamMessages.createMessage
+import schedule.IamMessages._
 import schedule.IamNotifier.createNotification
 import utils.attempt.FailedAttempt
 
 object IamAudit extends Logging {
 
-  def makeIamNotification(flaggedCreds: Map[AwsAccount, Seq[IAMAlertTargetGroup]]): List[IamNotification] = {
-    flaggedCreds.toList.flatMap { case (awsAccount, targetGroups) =>
+  def makeIamNotification(flaggedCredsToNotify: Map[AwsAccount, Seq[IAMAlertTargetGroup]]): List[IamNotification] = {
+    flaggedCredsToNotify.toList.flatMap { case (awsAccount, targetGroups) =>
       if (targetGroups.isEmpty) {
         logger.info(s"found no IAM user issues for ${awsAccount.name}. No notification required.")
         None
       } else {
         targetGroups.map { tg =>
-          logger.info(s"for ${awsAccount.name}, generating iam notification message for ${tg.users.length} user(s) with outdated keys and missing mfa")
-          tg.users.flatMap { user =>
-            val message = createMessage(tg.users, awsAccount)
-            Some(createNotification(awsAccount, tg.targets :+ Account(awsAccount.accountNumber), message, user.username, createDeadline(user.disableDeadline)))
+          logger.info(s"for ${awsAccount.name}, generating iam notification message for ${tg.users.length} user(s) with outdated keys and and/or missing mfa")
+
+          val (usersToReceiveWarningAlerts, usersToReceiveFinalAlerts) =
+            tg.users.foldRight[(Seq[VulnerableUser], Seq[VulnerableUser])]((Seq.empty, Seq.empty)) { case (user, (warningAcc, finalAcc)) =>
+              val deadline = createDeadline(user.disableDeadline)
+              if (isWarningAlert(deadline)) (user +: warningAcc, finalAcc)
+              else if (isFinalAlert(deadline)) (warningAcc, user +: finalAcc)
+              else (warningAcc, finalAcc)
+            }
+
+          val warningNotification = usersToReceiveWarningAlerts.map { user =>
+            val message = createWarningMessage(awsAccount, usersToReceiveWarningAlerts)
+            createNotification(awsAccount, tg.targets :+ Account(awsAccount.accountNumber), message, warningSubject(awsAccount), user.username, createDeadline(user.disableDeadline))
           }
+
+          val finalNotification = usersToReceiveFinalAlerts.map { user =>
+          val message = createFinalMessage(awsAccount, usersToReceiveWarningAlerts)
+          createNotification(awsAccount, tg.targets :+ Account(awsAccount.accountNumber), message, warningSubject(awsAccount), user.username, createDeadline(user.disableDeadline))
+           }
```

Comment:
the above two blocks look pretty similar - can they be simplified, maybe using a function?